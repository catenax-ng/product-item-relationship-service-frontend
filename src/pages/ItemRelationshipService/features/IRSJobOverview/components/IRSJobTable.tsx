import { Table } from "cx-portal-shared-components";
import { useTranslation } from "../../../../../lib";
// import { defaultDateFormat } from "../../../../../lib/dayjs";
import { JobStatusResult } from "../../../../../types/jobs";
import { IRSCancelJobButton } from "./IRSCancelJobButton";
import { IRSNavigateToJobDetails } from "./IRSNavigateToJobDetails";

export const IRSJobTable: React.FC<{
  isLoading: boolean;
  jobs: JobStatusResult[];
}> = ({ isLoading, jobs }) => {
  const { t } = useTranslation();
  const columns = [
    {
      field: "jobId",
      headerName: t("content.irs.jobsTable.jobID"),
      flex: 1,
      filterable: false,
    },
    // TODO: When backend adds data structure to data
    // {
    //   field: "start",
    //   headerName: t("content.irs.jobsTable.startDate"),
    //   width: 150,
    //   filterable: false,
    //   valueGetter: ({ row }: { row: JobStatusResult }) => defaultDateFormat(row.startedOn),
    // },
    // {
    //   field: "end",
    //   headerName: t("content.irs.jobsTable.endDate"),
    //   width: 150,
    //   filterable: false,
    //   valueGetter: ({ row }: { row: JobStatusResult }) => defaultDateFormat(row.jobCompleted),
    // },
    {
      field: "status",
      headerName: t("content.irs.jobsTable.status"),

      filterable: false,
      valueGetter: ({ row }: { row: JobStatusResult }) => row.status,
    },
    {
      field: "visualize",
      headerName: t("content.irs.jobsTable.visualize"),

      sortable: false,
      filterable: false,
      width: 150,
      renderCell: ({ row }: { row: JobStatusResult }) => {
        if (row.status === "RUNNING") {
          return <IRSCancelJobButton jobId={row.jobId} />;
        }
        if (row.status === "COMPLETED") {
          return <IRSNavigateToJobDetails jobId={row.jobId} />;
        }

        return null;
      },
    },
  ];
  return (
    <Table
      title={t("content.irs.jobsTable.title")}
      className="irs-table"
      columns={columns}
      rows={jobs}
      getRowId={(row: JobStatusResult) => {
        return row.jobId;
      }}
      loading={isLoading}
      disableColumnSelector={true}
      disableDensitySelector={true}
      hideFooter={true}
      disableColumnMenu={true}
    />
  );
};
