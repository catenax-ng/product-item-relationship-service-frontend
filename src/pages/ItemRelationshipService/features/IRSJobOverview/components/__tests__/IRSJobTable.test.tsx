import { MemoryRouter } from "react-router-dom";
import { JobStatusResultSuccess } from "../../../../../../services/api/__test__/jobStatusResult.example";
import { render } from "../../../../../../testing/test-utils";
import { IRSJobTable } from "../IRSJobTable";

it("renders a empty job table", () => {
  const { container } = render(
    <MemoryRouter>
      <IRSJobTable isLoading={false} jobs={[]} />
    </MemoryRouter>,
  );
  expect(container).toMatchSnapshot();
});

it("renders a with job variants", () => {
  const { container } = render(
    <MemoryRouter>
      <IRSJobTable isLoading={false} jobs={JobStatusResultSuccess} />
    </MemoryRouter>,
  );
  expect(container).toMatchSnapshot();
});
