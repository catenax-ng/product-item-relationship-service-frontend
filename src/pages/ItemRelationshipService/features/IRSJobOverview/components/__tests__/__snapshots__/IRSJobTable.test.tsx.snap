// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders a empty job table 1`] = `
<div>
  <div
    class="MuiBox-root css-b7ealj"
  >
    <div
      aria-colcount="3"
      aria-multiselectable="false"
      aria-rowcount="1"
      class="irs-table MuiDataGrid-root MuiDataGrid-autoHeight MuiDataGrid-root--densityStandard css-13itlwa-MuiDataGrid-root"
      role="grid"
    >
      <div>
        <div>
          <div
            class="MuiBox-root css-0"
          >
            <div
              class="MuiBox-root css-xgx8td"
            >
              <div
                class="MuiBox-root css-70qvj9"
              >
                <h5
                  class="MuiTypography-root MuiTypography-h5 css-ag7rrr-MuiTypography-root"
                >
                  content.irs.jobsTable.title
                  <span
                    class="MuiBox-root css-dah8a6"
                  >
                    (
                    0
                    )
                  </span>
                </h5>
              </div>
              <div
                class="MuiBox-root css-1bvc4cc"
              />
            </div>
          </div>
        </div>
      </div>
      <div
        class="MuiDataGrid-main css-204u17-MuiDataGrid-main"
      >
        <div
          style="height: auto; width: 0px; position: absolute; top: 57px; bottom: 0px; z-index: 4; pointer-events: none;"
        >
          <div
            class="MuiDataGrid-overlay css-1lk5ll6-MuiDataGrid-overlay"
          >
            No rows
          </div>
        </div>
        <div
          class="MuiDataGrid-columnHeaders css-f3jnds-MuiDataGrid-columnHeaders"
          style="min-height: 57px; max-height: 57px; line-height: 57px;"
        >
          <div
            aria-rowindex="1"
            class="MuiDataGrid-columnHeadersInner MuiDataGrid-columnHeadersInner--scrollable css-1s0hp0k-MuiDataGrid-columnHeadersInner"
            role="row"
            style="transform: translate3d(0px, 0px, 0px);"
          >
            <div
              aria-colindex="1"
              aria-sort="none"
              class="MuiDataGrid-columnHeader MuiDataGrid-columnHeader--sortable"
              data-field="jobId"
              role="columnheader"
              style="width: 0px; min-width: 0; max-width: 0;"
              tabindex="0"
            >
              <div
                class="MuiDataGrid-columnHeaderDraggableContainer"
                draggable="false"
              >
                <div
                  class="MuiDataGrid-columnHeaderTitleContainer"
                >
                  <div
                    class="MuiDataGrid-columnHeaderTitleContainerContent"
                  >
                    <div
                      aria-label=""
                      class="MuiDataGrid-columnHeaderTitle css-1jbbcbn-MuiDataGrid-columnHeaderTitle"
                      data-mui-internal-clone-element="true"
                    >
                      content.irs.jobsTable.jobID
                    </div>
                  </div>
                  <div
                    class="MuiDataGrid-iconButtonContainer css-ltf0zy-MuiDataGrid-iconButtonContainer"
                  >
                    <button
                      aria-label="Sort"
                      class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeSmall css-1pe4mpk-MuiButtonBase-root-MuiIconButton-root"
                      tabindex="-1"
                      title="Sort"
                      type="button"
                    >
                      <svg
                        aria-hidden="true"
                        class="MuiSvgIcon-root MuiSvgIcon-fontSizeSmall MuiDataGrid-sortIcon css-ptiqhd-MuiSvgIcon-root"
                        data-testid="ArrowUpwardIcon"
                        focusable="false"
                        viewBox="0 0 24 24"
                      >
                        <path
                          d="M4 12l1.41 1.41L11 7.83V20h2V7.83l5.58 5.59L20 12l-8-8-8 8z"
                        />
                      </svg>
                      <span
                        class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                      />
                    </button>
                  </div>
                </div>
              </div>
              <div
                class="MuiDataGrid-columnSeparator MuiDataGrid-columnSeparator--sideRight"
                style="min-height: 57px; opacity: 1;"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiDataGrid-iconSeparator css-i4bv87-MuiSvgIcon-root"
                  data-testid="SeparatorIcon"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M11 19V5h2v14z"
                  />
                </svg>
              </div>
            </div>
            <div
              aria-colindex="2"
              aria-sort="none"
              class="MuiDataGrid-columnHeader MuiDataGrid-columnHeader--sortable"
              data-field="status"
              role="columnheader"
              style="width: 100px; min-width: 100px; max-width: 100px;"
              tabindex="-1"
            >
              <div
                class="MuiDataGrid-columnHeaderDraggableContainer"
                draggable="false"
              >
                <div
                  class="MuiDataGrid-columnHeaderTitleContainer"
                >
                  <div
                    class="MuiDataGrid-columnHeaderTitleContainerContent"
                  >
                    <div
                      aria-label=""
                      class="MuiDataGrid-columnHeaderTitle css-1jbbcbn-MuiDataGrid-columnHeaderTitle"
                      data-mui-internal-clone-element="true"
                    >
                      content.irs.jobsTable.status
                    </div>
                  </div>
                  <div
                    class="MuiDataGrid-iconButtonContainer css-ltf0zy-MuiDataGrid-iconButtonContainer"
                  >
                    <button
                      aria-label="Sort"
                      class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeSmall css-1pe4mpk-MuiButtonBase-root-MuiIconButton-root"
                      tabindex="-1"
                      title="Sort"
                      type="button"
                    >
                      <svg
                        aria-hidden="true"
                        class="MuiSvgIcon-root MuiSvgIcon-fontSizeSmall MuiDataGrid-sortIcon css-ptiqhd-MuiSvgIcon-root"
                        data-testid="ArrowUpwardIcon"
                        focusable="false"
                        viewBox="0 0 24 24"
                      >
                        <path
                          d="M4 12l1.41 1.41L11 7.83V20h2V7.83l5.58 5.59L20 12l-8-8-8 8z"
                        />
                      </svg>
                      <span
                        class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                      />
                    </button>
                  </div>
                </div>
              </div>
              <div
                class="MuiDataGrid-columnSeparator MuiDataGrid-columnSeparator--sideRight"
                style="min-height: 57px; opacity: 1;"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiDataGrid-iconSeparator css-i4bv87-MuiSvgIcon-root"
                  data-testid="SeparatorIcon"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M11 19V5h2v14z"
                  />
                </svg>
              </div>
            </div>
            <div
              aria-colindex="3"
              aria-sort="none"
              class="MuiDataGrid-columnHeader"
              data-field="visualize"
              role="columnheader"
              style="width: 150px; min-width: 150px; max-width: 150px;"
              tabindex="-1"
            >
              <div
                class="MuiDataGrid-columnHeaderDraggableContainer"
                draggable="false"
              >
                <div
                  class="MuiDataGrid-columnHeaderTitleContainer"
                >
                  <div
                    class="MuiDataGrid-columnHeaderTitleContainerContent"
                  >
                    <div
                      aria-label=""
                      class="MuiDataGrid-columnHeaderTitle css-1jbbcbn-MuiDataGrid-columnHeaderTitle"
                      data-mui-internal-clone-element="true"
                    >
                      content.irs.jobsTable.visualize
                    </div>
                  </div>
                </div>
              </div>
              <div
                class="MuiDataGrid-columnSeparator MuiDataGrid-columnSeparator--sideRight"
                style="min-height: 57px; opacity: 1;"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiDataGrid-iconSeparator css-i4bv87-MuiSvgIcon-root"
                  data-testid="SeparatorIcon"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M11 19V5h2v14z"
                  />
                </svg>
              </div>
            </div>
          </div>
        </div>
        <div
          style="overflow: visible; width: 0px;"
        >
          <div
            class="MuiDataGrid-virtualScroller css-1w5m2wr-MuiDataGrid-virtualScroller"
            style="width: 0px; height: auto; margin-top: 57px; overflow-x: hidden; overflow-y: hidden;"
          >
            <div
              class="MuiDataGrid-virtualScrollerContent css-1kwdphh-MuiDataGrid-virtualScrollerContent"
              style="width: auto; height: 114px; min-height: auto;"
            >
              <div
                class="MuiDataGrid-virtualScrollerRenderZone css-s1v7zr-MuiDataGrid-virtualScrollerRenderZone"
                style="transform: translate3d(0px, undefinedpx, 0px);"
              />
            </div>
          </div>
        </div>
        <div
          class="Mui-resizeTriggers"
        >
          <div
            class="expand-trigger"
          >
            <div
              style="width: 1px; height: 1px;"
            />
          </div>
          <div
            class="contract-trigger"
          />
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`renders a with job variants 1`] = `
<div>
  <div
    class="MuiBox-root css-b7ealj"
  >
    <div
      aria-colcount="3"
      aria-multiselectable="false"
      aria-rowcount="5"
      class="irs-table MuiDataGrid-root MuiDataGrid-autoHeight MuiDataGrid-root--densityStandard css-13itlwa-MuiDataGrid-root"
      role="grid"
    >
      <div>
        <div>
          <div
            class="MuiBox-root css-0"
          >
            <div
              class="MuiBox-root css-xgx8td"
            >
              <div
                class="MuiBox-root css-70qvj9"
              >
                <h5
                  class="MuiTypography-root MuiTypography-h5 css-ag7rrr-MuiTypography-root"
                >
                  content.irs.jobsTable.title
                  <span
                    class="MuiBox-root css-dah8a6"
                  >
                    (
                    0
                    )
                  </span>
                </h5>
              </div>
              <div
                class="MuiBox-root css-1bvc4cc"
              />
            </div>
          </div>
        </div>
      </div>
      <div
        class="MuiDataGrid-main css-204u17-MuiDataGrid-main"
      >
        <div
          class="MuiDataGrid-columnHeaders css-f3jnds-MuiDataGrid-columnHeaders"
          style="min-height: 57px; max-height: 57px; line-height: 57px;"
        >
          <div
            aria-rowindex="1"
            class="MuiDataGrid-columnHeadersInner MuiDataGrid-columnHeadersInner--scrollable css-1s0hp0k-MuiDataGrid-columnHeadersInner"
            role="row"
            style="transform: translate3d(0px, 0px, 0px);"
          >
            <div
              aria-colindex="1"
              aria-sort="none"
              class="MuiDataGrid-columnHeader MuiDataGrid-columnHeader--sortable"
              data-field="jobId"
              role="columnheader"
              style="width: 0px; min-width: 0; max-width: 0;"
              tabindex="0"
            >
              <div
                class="MuiDataGrid-columnHeaderDraggableContainer"
                draggable="false"
              >
                <div
                  class="MuiDataGrid-columnHeaderTitleContainer"
                >
                  <div
                    class="MuiDataGrid-columnHeaderTitleContainerContent"
                  >
                    <div
                      aria-label=""
                      class="MuiDataGrid-columnHeaderTitle css-1jbbcbn-MuiDataGrid-columnHeaderTitle"
                      data-mui-internal-clone-element="true"
                    >
                      content.irs.jobsTable.jobID
                    </div>
                  </div>
                  <div
                    class="MuiDataGrid-iconButtonContainer css-ltf0zy-MuiDataGrid-iconButtonContainer"
                  >
                    <button
                      aria-label="Sort"
                      class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeSmall css-1pe4mpk-MuiButtonBase-root-MuiIconButton-root"
                      tabindex="-1"
                      title="Sort"
                      type="button"
                    >
                      <svg
                        aria-hidden="true"
                        class="MuiSvgIcon-root MuiSvgIcon-fontSizeSmall MuiDataGrid-sortIcon css-ptiqhd-MuiSvgIcon-root"
                        data-testid="ArrowUpwardIcon"
                        focusable="false"
                        viewBox="0 0 24 24"
                      >
                        <path
                          d="M4 12l1.41 1.41L11 7.83V20h2V7.83l5.58 5.59L20 12l-8-8-8 8z"
                        />
                      </svg>
                      <span
                        class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                      />
                    </button>
                  </div>
                </div>
              </div>
              <div
                class="MuiDataGrid-columnSeparator MuiDataGrid-columnSeparator--sideRight"
                style="min-height: 57px; opacity: 1;"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiDataGrid-iconSeparator css-i4bv87-MuiSvgIcon-root"
                  data-testid="SeparatorIcon"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M11 19V5h2v14z"
                  />
                </svg>
              </div>
            </div>
            <div
              aria-colindex="2"
              aria-sort="none"
              class="MuiDataGrid-columnHeader MuiDataGrid-columnHeader--sortable"
              data-field="status"
              role="columnheader"
              style="width: 100px; min-width: 100px; max-width: 100px;"
              tabindex="-1"
            >
              <div
                class="MuiDataGrid-columnHeaderDraggableContainer"
                draggable="false"
              >
                <div
                  class="MuiDataGrid-columnHeaderTitleContainer"
                >
                  <div
                    class="MuiDataGrid-columnHeaderTitleContainerContent"
                  >
                    <div
                      aria-label=""
                      class="MuiDataGrid-columnHeaderTitle css-1jbbcbn-MuiDataGrid-columnHeaderTitle"
                      data-mui-internal-clone-element="true"
                    >
                      content.irs.jobsTable.status
                    </div>
                  </div>
                  <div
                    class="MuiDataGrid-iconButtonContainer css-ltf0zy-MuiDataGrid-iconButtonContainer"
                  >
                    <button
                      aria-label="Sort"
                      class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeSmall css-1pe4mpk-MuiButtonBase-root-MuiIconButton-root"
                      tabindex="-1"
                      title="Sort"
                      type="button"
                    >
                      <svg
                        aria-hidden="true"
                        class="MuiSvgIcon-root MuiSvgIcon-fontSizeSmall MuiDataGrid-sortIcon css-ptiqhd-MuiSvgIcon-root"
                        data-testid="ArrowUpwardIcon"
                        focusable="false"
                        viewBox="0 0 24 24"
                      >
                        <path
                          d="M4 12l1.41 1.41L11 7.83V20h2V7.83l5.58 5.59L20 12l-8-8-8 8z"
                        />
                      </svg>
                      <span
                        class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                      />
                    </button>
                  </div>
                </div>
              </div>
              <div
                class="MuiDataGrid-columnSeparator MuiDataGrid-columnSeparator--sideRight"
                style="min-height: 57px; opacity: 1;"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiDataGrid-iconSeparator css-i4bv87-MuiSvgIcon-root"
                  data-testid="SeparatorIcon"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M11 19V5h2v14z"
                  />
                </svg>
              </div>
            </div>
            <div
              aria-colindex="3"
              aria-sort="none"
              class="MuiDataGrid-columnHeader"
              data-field="visualize"
              role="columnheader"
              style="width: 150px; min-width: 150px; max-width: 150px;"
              tabindex="-1"
            >
              <div
                class="MuiDataGrid-columnHeaderDraggableContainer"
                draggable="false"
              >
                <div
                  class="MuiDataGrid-columnHeaderTitleContainer"
                >
                  <div
                    class="MuiDataGrid-columnHeaderTitleContainerContent"
                  >
                    <div
                      aria-label=""
                      class="MuiDataGrid-columnHeaderTitle css-1jbbcbn-MuiDataGrid-columnHeaderTitle"
                      data-mui-internal-clone-element="true"
                    >
                      content.irs.jobsTable.visualize
                    </div>
                  </div>
                </div>
              </div>
              <div
                class="MuiDataGrid-columnSeparator MuiDataGrid-columnSeparator--sideRight"
                style="min-height: 57px; opacity: 1;"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiDataGrid-iconSeparator css-i4bv87-MuiSvgIcon-root"
                  data-testid="SeparatorIcon"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M11 19V5h2v14z"
                  />
                </svg>
              </div>
            </div>
          </div>
        </div>
        <div
          style="overflow: visible; width: 0px;"
        >
          <div
            class="MuiDataGrid-virtualScroller css-1w5m2wr-MuiDataGrid-virtualScroller"
            style="width: 0px; height: auto; margin-top: 57px; overflow-x: hidden; overflow-y: hidden;"
          >
            <div
              class="MuiDataGrid-virtualScrollerContent css-1kwdphh-MuiDataGrid-virtualScrollerContent"
              style="width: auto; height: 228px; min-height: auto;"
            >
              <div
                class="MuiDataGrid-virtualScrollerRenderZone css-s1v7zr-MuiDataGrid-virtualScrollerRenderZone"
                style="transform: translate3d(0px, 0px, 0px);"
              >
                <div
                  aria-rowindex="2"
                  aria-selected="false"
                  class="MuiDataGrid-row"
                  data-id="demo-initial"
                  data-rowindex="0"
                  role="row"
                  style="max-height: 57px; min-height: 57px;"
                >
                  <div
                    aria-colindex="1"
                    aria-colspan="1"
                    class="MuiDataGrid-cell MuiDataGrid-cell--textLeft"
                    data-colindex="0"
                    data-field="jobId"
                    role="cell"
                    style="min-width: 0; max-width: 0; min-height: 57px; max-height: 57px;"
                    tabindex="-1"
                  >
                    <div
                      class="MuiDataGrid-cellContent"
                    >
                      demo-initial
                    </div>
                  </div>
                  <div
                    aria-colindex="2"
                    aria-colspan="1"
                    class="MuiDataGrid-cell MuiDataGrid-cell--textLeft"
                    data-colindex="1"
                    data-field="status"
                    role="cell"
                    style="min-width: 100px; max-width: 100px; min-height: 57px; max-height: 57px;"
                    tabindex="-1"
                  >
                    <div
                      class="MuiDataGrid-cellContent"
                    >
                      INITIAL
                    </div>
                  </div>
                  <div
                    aria-colindex="3"
                    aria-colspan="1"
                    class="MuiDataGrid-cell--withRenderer MuiDataGrid-cell MuiDataGrid-cell--textLeft"
                    data-colindex="2"
                    data-field="visualize"
                    role="cell"
                    style="min-width: 150px; max-width: 150px; min-height: 57px; max-height: 57px;"
                    tabindex="-1"
                  >
                    <div
                      class="MuiDataGrid-cellContent"
                    />
                  </div>
                </div>
                <div
                  aria-rowindex="3"
                  aria-selected="false"
                  class="MuiDataGrid-row"
                  data-id="demo-running"
                  data-rowindex="1"
                  role="row"
                  style="max-height: 57px; min-height: 57px;"
                >
                  <div
                    aria-colindex="1"
                    aria-colspan="1"
                    class="MuiDataGrid-cell MuiDataGrid-cell--textLeft"
                    data-colindex="0"
                    data-field="jobId"
                    role="cell"
                    style="min-width: 0; max-width: 0; min-height: 57px; max-height: 57px;"
                    tabindex="-1"
                  >
                    <div
                      class="MuiDataGrid-cellContent"
                    >
                      demo-running
                    </div>
                  </div>
                  <div
                    aria-colindex="2"
                    aria-colspan="1"
                    class="MuiDataGrid-cell MuiDataGrid-cell--textLeft"
                    data-colindex="1"
                    data-field="status"
                    role="cell"
                    style="min-width: 100px; max-width: 100px; min-height: 57px; max-height: 57px;"
                    tabindex="-1"
                  >
                    <div
                      class="MuiDataGrid-cellContent"
                    >
                      RUNNING
                    </div>
                  </div>
                  <div
                    aria-colindex="3"
                    aria-colspan="1"
                    class="MuiDataGrid-cell--withRenderer MuiDataGrid-cell MuiDataGrid-cell--textLeft"
                    data-colindex="2"
                    data-field="visualize"
                    role="cell"
                    style="min-width: 150px; max-width: 150px; min-height: 57px; max-height: 57px;"
                    tabindex="-1"
                  >
                    <button
                      class="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorSecondary MuiIconButton-sizeSmall css-1hxb263-MuiButtonBase-root-MuiIconButton-root"
                      style="align-self: center;"
                      tabindex="0"
                      type="button"
                    >
                      <svg
                        aria-hidden="true"
                        class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                        data-testid="CancelIcon"
                        focusable="false"
                        viewBox="0 0 24 24"
                      >
                        <path
                          d="M12 2C6.47 2 2 6.47 2 12s4.47 10 10 10 10-4.47 10-10S17.53 2 12 2zm5 13.59L15.59 17 12 13.41 8.41 17 7 15.59 10.59 12 7 8.41 8.41 7 12 10.59 15.59 7 17 8.41 13.41 12 17 15.59z"
                        />
                      </svg>
                      <span
                        class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                      />
                    </button>
                  </div>
                </div>
                <div
                  aria-rowindex="4"
                  aria-selected="false"
                  class="MuiDataGrid-row"
                  data-id="demo-completed"
                  data-rowindex="2"
                  role="row"
                  style="max-height: 57px; min-height: 57px;"
                >
                  <div
                    aria-colindex="1"
                    aria-colspan="1"
                    class="MuiDataGrid-cell MuiDataGrid-cell--textLeft"
                    data-colindex="0"
                    data-field="jobId"
                    role="cell"
                    style="min-width: 0; max-width: 0; min-height: 57px; max-height: 57px;"
                    tabindex="-1"
                  >
                    <div
                      class="MuiDataGrid-cellContent"
                    >
                      demo-completed
                    </div>
                  </div>
                  <div
                    aria-colindex="2"
                    aria-colspan="1"
                    class="MuiDataGrid-cell MuiDataGrid-cell--textLeft"
                    data-colindex="1"
                    data-field="status"
                    role="cell"
                    style="min-width: 100px; max-width: 100px; min-height: 57px; max-height: 57px;"
                    tabindex="-1"
                  >
                    <div
                      class="MuiDataGrid-cellContent"
                    >
                      COMPLETED
                    </div>
                  </div>
                  <div
                    aria-colindex="3"
                    aria-colspan="1"
                    class="MuiDataGrid-cell--withRenderer MuiDataGrid-cell MuiDataGrid-cell--textLeft"
                    data-colindex="2"
                    data-field="visualize"
                    role="cell"
                    style="min-width: 150px; max-width: 150px; min-height: 57px; max-height: 57px;"
                    tabindex="-1"
                  >
                    <a
                      href="/jobs/DEV/demo-completed"
                    >
                      <button
                        class="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorSecondary MuiIconButton-sizeSmall css-1hxb263-MuiButtonBase-root-MuiIconButton-root"
                        style="align-self: center;"
                        tabindex="0"
                        type="button"
                      >
                        <svg
                          aria-hidden="true"
                          class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                          data-testid="ArrowForwardIcon"
                          focusable="false"
                          viewBox="0 0 24 24"
                        >
                          <path
                            d="m12 4-1.41 1.41L16.17 11H4v2h12.17l-5.58 5.59L12 20l8-8z"
                          />
                        </svg>
                        <span
                          class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                        />
                      </button>
                    </a>
                  </div>
                </div>
                <div
                  aria-rowindex="5"
                  aria-selected="false"
                  class="MuiDataGrid-row MuiDataGrid-row--lastVisible"
                  data-id="demo-canceled"
                  data-rowindex="3"
                  role="row"
                  style="max-height: 57px; min-height: 57px;"
                >
                  <div
                    aria-colindex="1"
                    aria-colspan="1"
                    class="MuiDataGrid-cell MuiDataGrid-cell--textLeft"
                    data-colindex="0"
                    data-field="jobId"
                    role="cell"
                    style="min-width: 0; max-width: 0; min-height: 57px; max-height: 57px;"
                    tabindex="-1"
                  >
                    <div
                      class="MuiDataGrid-cellContent"
                    >
                      demo-canceled
                    </div>
                  </div>
                  <div
                    aria-colindex="2"
                    aria-colspan="1"
                    class="MuiDataGrid-cell MuiDataGrid-cell--textLeft"
                    data-colindex="1"
                    data-field="status"
                    role="cell"
                    style="min-width: 100px; max-width: 100px; min-height: 57px; max-height: 57px;"
                    tabindex="-1"
                  >
                    <div
                      class="MuiDataGrid-cellContent"
                    >
                      CANCELED
                    </div>
                  </div>
                  <div
                    aria-colindex="3"
                    aria-colspan="1"
                    class="MuiDataGrid-cell--withRenderer MuiDataGrid-cell MuiDataGrid-cell--textLeft"
                    data-colindex="2"
                    data-field="visualize"
                    role="cell"
                    style="min-width: 150px; max-width: 150px; min-height: 57px; max-height: 57px;"
                    tabindex="-1"
                  >
                    <div
                      class="MuiDataGrid-cellContent"
                    />
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
        <div
          class="Mui-resizeTriggers"
        >
          <div
            class="expand-trigger"
          >
            <div
              style="width: 1px; height: 1px;"
            />
          </div>
          <div
            class="contract-trigger"
          />
        </div>
      </div>
    </div>
  </div>
</div>
`;
